src/utils/restApiWrapper.ts

```diff
@@ -0,0 +1,58 @@
+export async function postData(
+    url = '',
+    headers = {},
+    data = {},
+): Promise<any> {
+    const response = await fetch(url, {
+        method: 'POST',
+        mode: 'no-cors',
+        headers: {
+            ...headers,
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(data),
+    })
+    return response
+}
+
+export async function getData(url = '', headers = {}): Promise<any> {
+    const response = await fetch(url, {
+        method: 'GET',
+        mode: 'no-cors',
+        headers: {
+            ...headers,
+            'Content-Type': 'application/json',
+        },
+    })
+    return response
+}
+
+export async function putData(url = '', headers = {}, data = {}): Promise<any> {
+    const response = await fetch(url, {
+        method: 'PUT',
+        mode: 'no-cors',
+        headers: {
+            ...headers,
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(data),
+    })
+    return response
+}
+
+export async function deleteData(
+    url = '',
+    headers = {},
+    data = {},
+): Promise<any> {
+    const response = await fetch(url, {
+        method: 'DELETE',
+        mode: 'no-cors',
```

Comment:
Isn't this unsafe?

Reply:
As of now, yes, we will do away with `no-cors` once the updated endpoints are up and we allow-list endpoints originating from our backend.

Reply:
Done, removed the `no-cors` mode.